perma_web/perma/tests/test_views_user_settings.py

```diff
@@ -0,0 +1,648 @@
+        data={'account_type':'Registrar'}
+    )
+
+    assert user.registrar.name.encode() in response.content
+    assert b'Personal' not in response.content
+
+
+#
+# Tools
+#
+
+def test_api_key(client, link_user):
+
+    def get_api_key():
+        response = client.get(reverse('settings_tools'), secure=True)
+        assert response.status_code == 200
+        soup = BeautifulSoup(response.content, 'html.parser')
+        api_key_input = soup.find('input', {'id': 'id_api_key'})
+        if api_key_input:
+            key = api_key_input.get('value', None)
+        else:
+            key = None
+        return key
+
+
+    def create_api_key():
+        submit_form(
+            client,
+            'api_key_create',
+            success_url=reverse('settings_tools')
+        )
+
+    client.force_login(link_user)
+
+    # No API key is present at first
+    key = get_api_key()
+    assert not key
+
+    # We can generate one
+    create_api_key()
+    first_key = get_api_key()
+    assert first_key
+
+    # We can generate a new one
+    create_api_key()
+    second_key = get_api_key()
+    assert second_key
+    assert first_key != second_key
+
+
+#
+# Affiliations: Does the expected information show up on the affiliations page?
+#
+
+def get_afffilation_markup_for_user(client, user):
+    client.force_login(user)
+    response = client.get(reverse('settings_affiliations'), secure=True)
+    return response, BeautifulSoup(response.content, 'html.parser')
+
+
+def test_affiliations_org_user(client, multi_registrar_org_user):
+    registrar_names = [org.registrar.name for org in multi_registrar_org_user.organizations.all()]
+    org_names = [org.name for org in multi_registrar_org_user.organizations.all()]
+
+    _, soup = get_afffilation_markup_for_user(client, multi_registrar_org_user)
+
+    registrars = soup.select('h4 a')
+    assert len(registrars) == 2
+    for registrar in registrars:
+        assert registrar.text.strip() in registrar_names
+
+    orgs = soup.select('.settings-block p')
+    assert len(orgs) == 2
+    for org in orgs:
+        assert org.text.strip() in org_names
+
+
+def test_affiliations_registrar_user(client, registrar_user):
+    _, soup = get_afffilation_markup_for_user(client, registrar_user)
+
+    [registrar] = soup.select('h4')
+    assert registrar.text.strip() == registrar_user.registrar.name
+
+    registrar_settings = soup.select('dt')
+    assert len(registrar_settings) == 2
+    for setting in registrar_settings:
+        assert setting.text.strip() in ["Website", "Email"]
+
+
+def test_affiliations_pending_registrar_user(client, pending_registrar_user):
+    response, soup = get_afffilation_markup_for_user(client, pending_registrar_user)
+
+    assert b'Pending Registrar'in response.content
+    assert b'Thank you for requesting an account for your library. Perma.cc will review your request as soon as possible.' in response.content
+
+    [registrar] = soup.select('.sponsor-name')
```

Comment:
See previous comment. Optional suggestion:
```suggestion
    registrar = soup.select_one('.sponsor-name')
```